libp2p/peerinfo.nim

```diff
@@ -39,6 +39,7 @@ type
       privateKey*: PrivateKey
     of HasPublic:
       key: Option[PublicKey]
+    seqno*: uint64
```

Comment:
storing this in the peerinfo looks odd - it means we'll keep a copy for every peer out there

Reply:
it should definitely not be here, `PeerInfo` is a generic object that doesn't know anything about `seqno`, in addition to what @arnetheduck it is also a leaking abstraction.

Reply:
Should/Could be for each peer tho, 
I'm fine with adding a static global tho.. or thread local if fine for you (i'd prefer even)